dora/core/common/src/main/java/alluxio/wire/WorkerIdentity.java

```diff
@@ -360,4 +373,138 @@ public UUID toUUID(WorkerIdentity identity) throws IllegalArgumentException {
       return getVersionSpecificRepresentation0(identity);
     }
   }
+
+  /**
+   * Utility for ser/de with Gson.
+   * <p>
+   * The version field is encoded as a JSON number, and the identity field
+   * is encoded as a hexadecimal string.
+   */
+  public enum GsonSerde
```

Comment:
can u change name to GsonSerializer?

Reply:
is this serializer too much? why do we need such enum for serialize 2 fields?

Reply:
this is both a serializer and a deserializer.

> why do we need such enum for serialize 2 fields?

Most of the code is in the deserializer, which needs to handle cases like incorrect type of fields, fields missing, etc.